src/py/flwr/common/serde.py

```diff
@@ -751,3 +752,70 @@ def recordset_from_proto(recordset_proto: ProtoRecordSet) -> RecordSet:
             k: configs_record_from_proto(v) for k, v in recordset_proto.configs.items()
         },
     )
+
+
+# === Message ===
+
+
+def message_to_task_ins(message: Message) -> TaskIns:
+    """Create a TaskIns from the Message."""
+    return TaskIns(
+        task_id="",  # This will be generated by the server
+        group_id=message.metadata.group_id,
+        run_id=message.metadata.run_id,
+        task=Task(
+            ttl=message.metadata.ttl,
+            task_type=message.metadata.task_type,
+            recordset=recordset_to_proto(message.message),
+        ),
+    )
+
+
+def message_from_task_ins(task_ins: TaskIns) -> Message:
+    """Create a Message from the TaskIns."""
+    # Retrieve the Metadata
+    metadata = Metadata(
+        run_id=task_ins.run_id,
+        task_id=task_ins.task_id,
+        group_id=task_ins.group_id,
+        ttl=task_ins.task.ttl,
+        task_type=task_ins.task.task_type,
+    )
+
+    # Return the Message
+    return Message(
+        metadata=metadata,
+        message=recordset_from_proto(task_ins.task.recordset),
+    )
+
+
+def message_to_task_res(message: Message) -> TaskRes:
+    """Create a TaskRes from the Message."""
+    return TaskRes(
+        task_id="",  # This will be generated by the server
+        group_id=message.metadata.group_id,
+        run_id=message.metadata.run_id,
```

Comment:
We will probably take these from the `TaskIns` instead of the `Message` we get from the `FlowerCallable`

Reply:
Yes, make sense. These fields should be set by the framework instead of being taken from `Message.Metadata`.